Lib/test/test_bytes.py

```diff
@@ -1050,7 +1050,7 @@ class A:
             def __bytes__(self):
                 return OtherBytesSubclass(b'abc')
         self.assertEqual(bytes(A()), b'abc')
-        self.assertIs(type(bytes(A())), OtherBytesSubclass)
+        self.assertIs(type(bytes(A())), bytes)
```

Comment:
The test being changed here was added in Python 3.4, in https://github.com/python/cpython/commit/15095800a381a396cbc077cb5320203a8feae51a (by @serhiy-storchaka). The commit was a fix for issue #68919.

(I haven't thought deeply about whether the change here is correct or desirable; just providing context.)

Reply:
The test was added to cover all paths, so the behavior change could not be unnoticed. I think that the new behavior is more correct.

But, for consistency with numeric types there should be a warning here. It may be a separate PR.

Reply:
@serhiy-storchaka Do you think we should go on with it, or wait for more opinions?